src/Symfony/Component/HttpKernel/Attribute/MapQueryParameter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+/**
+ * Can be used in a controller action to mark a parameter as a query parameter.
+ *
+ * public function homeAction(
+ *   #[MapQueryParameter] ?string $name
+ * );
+ *
+ * When opening this URL: /home?name=John
+ * It will fill the $name parameter with the value of the query parameter "name".
+ *
+ * When opening this URL: /home
+ * It will set $name to null.
+ *
+ * When the parameter is non-nullable / not optional:
+ * public function homeAction(
+ *   #[MapQueryParameter] string $name
+ * );
+ *
+ * Opening /home without passing ?name=John will throw a Bad Request exception.
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class MapQueryParameter
+{
+    /**
+     * @param string|null $name The name of the query parameter. If null, the name of the parameter in the action will be used.
+     */
+    public function __construct(
+        public ?string $name = null,
```

Comment:
What about adding $options as allowed by https://www.php.net/manual/en/filter.filters.validate.php?

I'd suggest compacting options+flags+filter this way:
`#[MapQueryString(options: ['filter' => FILTER_VALIDATE_REGEXP, 'regexp' => '/abc/', 'flags' FILTER_NULL_ON_FAILURE,])`

Reply:
Let me give this a try :)

Reply:
Is this what you head in mind? See last 2 commits. The last one is something that we need to think about,.... 🤔

Reply:
Fixed.